Add basic component and partial benchmarks (#452)
github/view_component: README.md

ViewComponents use a standard Ruby initializer that clearly defines what is needed to render, making them easier (and safer) to reuse than partials.

#### Performance

> Based on our benchmarks, components are 10x faster than partials. You can find the benchmark script we used to get that number [here](performance/benchmark.rb). If you'd like to run the benchmarks locally, you can run `bundle exec rake benchmark`.

Reviewer: ```suggestion
Based on our [benchmarks](performance/benchmark.rb), ViewComponents are ~10x faster than partials. Run `bundle exec rake benchmark` to see for yourself!
```

> The benchmark script is a work-in-progress, so improvements are welcome!
> 

Reviewer: ```suggestion
```
